src/otx/algo/classification/otx_dino_v2.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""DINO-V2 model for the OTX classification."""
+
+from __future__ import annotations
+
+from collections import OrderedDict
+from typing import TYPE_CHECKING, Any
+
+import torch
+from torch import nn
+
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.classification import MulticlassClsBatchDataEntity, MulticlassClsBatchPredEntity
+from otx.core.model.entity.classification import OTXClassificationModel
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+class DINOv2RegisterClassifier(OTXClassificationModel):
+    """DINO-v2 Classification Model."""
+    def __init__(self, config: DictConfig) -> None:
+        self.config = config
+        super().__init__() # create the model
+
+        self.loss = nn.CrossEntropyLoss()
+
+        # NOTE,
+        # We've decided to use MMpretrain's pipeline for this model
+        # It's hard to use ClsDataPreprocessor since the model is not related to MMpretrain
+        # That's the reason why I implemented the below preprocess things
+        self.data_preprocess_cfg = self.config.data_preprocess
+        self.register_buffer(
+            'mean', torch.tensor(self.data_preprocess_cfg.mean).view(-1, 1, 1), False,
+        )
+        self.register_buffer(
+            'std', torch.tensor(self.data_preprocess_cfg.std).view(-1, 1, 1), False,
+        )
+
+    def _create_model(self) -> nn.Module:
+        """Create the model."""
+        self.backbone = torch.hub.load(
+            repo_or_dir="facebookresearch/dinov2",
+            model=self.config.backbone.name,
+        )
+        if self.config.backbone.frozen:
+            self._freeze_backbone(self.backbone)
+
+        self.head = nn.Linear(
+            self.config.head.in_channels,
+            self.config.head.num_classes,
+        )
+
+        return nn.Sequential(
+            OrderedDict([
+                ("backbone", self.backbone),
+                ("head", self.head),
+            ]),
+        )
+
+    def _freeze_backbone(self, backbone: nn.Module) -> None:
+        """Freeze the backbone."""
+        for _, v in backbone.named_parameters():
+            v.requires_grad = False
+
+    def _preprocess_img(self, imgs: torch.Tensor) -> torch.Tensor:
+        """Control normalize and BGR/RGB conversion."""
+        # BGR -> RGB
+        if self.data_preprocess_cfg.to_rgb and imgs.size(1) == 3:
+            imgs = imgs.flip(1)
+
+        # Normalization
+        imgs = imgs.float()
```

Comment:
Please do this at initialization.

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2708/commits/7c8db3a54e84a4e183a48378fdd428e047288441